### PR TITLE
Zombie Rotation Locking Fix

### PR DESCRIPTION
**Pull request by Cojoke-dot**
## About the PR

Fixes zombie rotation getting stuck all the time
## Why / Balance

Fix zombie bug, also fixes a bug I found in #30478. If a player has a ghost role and then leaves their body with combat mode on it locks the NPCs rotation.
## Technical details

Remove the components for mouse rotation when a player leaves a body that has an NPC in it. Moves the combat mode stuff for zombification after it adds the NPC stuff so players that have ghosted count as `IsNpc` in `SetInCombatMode`.
## Media

https://github.com/user-attachments/assets/8ab051a8-762f-40c1-8609-2d894c63535f

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Fix zombie NPCs rotation getting locked sometimes

### CODE REVIEW

**Comment by slarticodefast** [comment]
Can you fix the merge conflict?

**Comment by lzk228** [comment]
fixes https://github.com/space-wizards/space-station-14/issues/28048

**Comment by Cojoke-dot** [comment]
> Can you fix the merge conflict? 

Yeah sure, I'll be busy for a couple of days so it may be a little bit.

**Comment by slarticodefast** [comment]
Did some testing and there seems to still be a bug with the mouse rotator in there.
- spawn an Urist
- zombify yourself
- kill and turn the Urist into a zombie
- /ghost
- take the zombie ghost roll
- you will start in combat mode, but the mouse rotator is missing, meaning you won't point towards your mouse
- deactivating and reactivating combat mode fixes it

File: Content.Server/NPC/Systems/NPCSystem.cs [hunk 1]
```diff
@@ -55,6 +57,8 @@ public void OnPlayerNPCDetach(EntityUid uid, HTNComponent component, PlayerDetac
             if (TryComp<MindContainerComponent>(uid, out var mindContainer) && mindContainer.HasMind)
                 return;
 
+            RemComp<MouseRotatorComponent>(uid);
+            RemComp<NoRotateOnMoveComponent>(uid);
             WakeNPC(uid, component);
         }
```
Comment (metalgearsloth) [inline Content.Server/NPC/Systems/NPCSystem.cs:63]: This will break if they leave and go back in, better to just have combat mode toggled.

**Pull request closed without merging**